tests/knr1050.py

```diff
@@ -0,0 +1,211 @@
+import unittest
+
+from utils.channel_access import ChannelAccess
+from utils.ioc_launcher import get_default_ioc_dir
+from utils.test_modes import TestModes
+from utils.testing import get_running_lewis_and_ioc, skip_if_recsim
+
+DEVICE_PREFIX = "KNR1050_01"
+
+device_name = "knr1050"
+
+IOCS = [
+    {
+        "name": DEVICE_PREFIX,
+        "directory": get_default_ioc_dir("KNR1050"),
+        "macros": {},
+        "emulator": device_name,
+    },
+]
+
+
+TEST_MODES = [TestModes.DEVSIM, TestModes.RECSIM]
+
+
+class Knr1050Tests(unittest.TestCase):
+    """
+    Tests for the Knr1050 IOC.
+    """
+    def setUp(self):
```

Comment:
This is a large function and is confusing why you are setting these values. Refactor common ones out into a common function and name the function well.